### ISSUE TITLE
Can we use Page Factory with ngWebDriver?

### ISSUE THREAD

**Issue by Niks-Dev** [paul-hammant/ngWebDriver#26]
Just like @FindBy annotation for locators(id,css,name,xpath etc) in  Selenium , do we have any facility to use angular locators(model,binding etc) in Page Factory??

Thanks in advance.

**Comment by paul-hammant** [comment]
Show a me code snippet of what you want to do, please :)

**Comment by Niks-Dev** [comment]
Hi Paul,

Thanks for your reply. Please find code snippet below:- 

Below code of Selenium,
**driver.findElement(By.id("username")).sendKeys("Paul");**

will be look like this after using pagefactory

**pf_instance.Username.sendKeys("Paul");**

Locator store like this in PageFactory
@FindBy(id = "username")
public WebElement Username;

SImilarly can we do something angular locators as well? Annotation like @FindByNG

Please see this for PageFactory reference :-
http://www.guru99.com/page-object-model-pom-page-factory-in-selenium-ultimate-guide.html
http://toolsqa.com/selenium-webdriver/page-object-pattern-model-page-factory/

Thanks in advance.

**Comment by paul-hammant** [comment]
I'll see what I can do.

**Comment by Niks-Dev** [comment]
Thanks Paul, it would be great help.

**Comment by adityagautam** [comment]
Hey Paul. Do we have any timeframe when can we expect this feature/enhancement in ngwebdriver.

**Comment by patota** [comment]
Hello.
News for this issue? :)

**Comment by paul-hammant** [comment]
Discussion started with selenium developers - https://groups.google.com/forum/#!topic/selenium-developers/_ts_NrrteWI

**Comment by Surxon08** [comment]
Hi Paul!

are there any news regarding this question?

Thank you

**Comment by Nandesh7** [comment]
@paul-hammant I read your changes are committed to https://github.com/SeleniumHQ/selenium/commit/0b3018895b4cd1f6e7f398b98ecc8ce4cdd877a3. Could you please share a sample test case that relies on pagefactory using Angular? Thanks

**Comment by Nandesh7** [comment]
@paul-hammant  I saw a similar implementation here https://github.com/henrrich/jpagefactory

**Comment by paul-hammant** [comment]
Good catch - I did all the code to prove it works but did not commit it anywhere. I'll find time to complete :)

**Comment by Nandesh7** [comment]
@paul-hammant That sounds like a plan. Do you have a timeline for it? Thanks

**Comment by paul-hammant** [comment]
Take a look at https://github.com/paul-hammant/ngWebDriver/blob/master/src/test/java/com/paulhammant/ngwebdriver/FindByNgWebDriverTest.java lines 14 through 45.  Is that what you're looking for @Nandesh7 and @Niks-Dev ?

**Comment by paul-hammant** [comment]
v1.1.2 released
